README.md

```diff
@@ -1349,6 +1349,31 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
 * <a id='newline-at-eof'></a>(<a href='#newline-at-eof'>link</a>) **Files should end in a newline.** [![SwiftLint: trailing_newline](https://img.shields.io/badge/SwiftLint-trailing__newline-007A87.svg)](https://github.com/realm/SwiftLint/blob/master/Rules.md#trailing-newline)
 
+* <a id='mark-types-and-extensions'></a>(<a href='#mark-types-and-extensions'>link</a>) **Each type and extension should be preceded by a `MARK` comment.** [![SwiftFormat: markTypes](https://img.shields.io/badge/SwiftFormat-markTypes-008489.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#markTypes)
+  * Types should be preceded by a `// MARK: - TypeName` comment.
+  * Extensions that add a conformance should be preceded by a `// MARK: TypeName + ProtocolName` comment.
```

Comment:
Shouldn't this also have a dash since it's a standalone type and it should have its own section?
```suggestion
  * Extensions that add a conformance should be preceded by a `// MARK: - TypeName + ProtocolName` comment.
```

Reply:
Hmm good question. Do we want the rule to prefer using a dash _unless_ it's an extension following the type being extended?

The SwiftFormat rule is currently set up to support different comment templates for types (`// MARK: - %t` by default) and extensions (`// MARK: %t` by default). I would have to rethink how the SwiftFormat rule works to support this behavior, but am open to doing that if this is the direction we want to go in.

Reply:
I think it makes the most sense if we use `-` to indicate separation between types, so I agree with Eric's revision.

Reply:
https://github.com/nicklockwood/SwiftFormat/pull/769 adds support for this behavior to the SwiftFormat `markTypes` rule.